src/napistu/mechanism_matching.py

```diff
@@ -1,28 +1,121 @@
 from __future__ import annotations
 
 import logging
+from typing import Optional, Union, Set, Dict, List
 
 import igraph as ig
+import numpy as np
 import pandas as pd
+
+from napistu import identifiers
 from napistu import sbml_dfs_core
 from napistu import utils
 from napistu.constants import SBML_DFS
 from napistu.constants import CPR_EDGELIST
 from napistu.constants import CPR_EDGELIST_REQ_VARS
+from napistu.constants import FEATURE_ID_VAR_DEFAULT
+from napistu.constants import RESOLVE_MATCHES_AGGREGATORS
+from napistu.constants import RESOLVE_MATCHES_TMP_WEIGHT_COL
 from napistu.constants import IDENTIFIERS
 from napistu.constants import IDENTIFIER_EDGELIST_REQ_VARS
-from napistu.constants import SPECIES_IDENTIFIERS_REQUIRED_VARS
+from napistu.constants import ONTOLOGIES_LIST
 from napistu.network.constants import CPR_GRAPH_EDGES
 from napistu.network import paths
 
 logger = logging.getLogger(__name__)
 
 
+def bind_wide_results(
```

Comment:
This seems one of the functions doing the heavy lifting.

One of the crucial steps of the matching are matching the features from the **results_df** to the **smbl_dfs** species ids.

I am thinking if it would be feasible to add an optional argument to allow to provide a "manual_matching" that is a dict going from 'feature_id" to the list of species_ids. This would enable the user to also more easily implement some matching strategies.
Eg it could be interesting to have a prioritized matching: provide an ordered list of ontologies match based on the first match of an ontolgy. This would be useful in cases where some ontologies are broader than others and we want to get the most specific match given the ontologies.

Reply:
On a second glance:
The function is already broken down into nice substeps (`_prepare_species_identifiers`, `match...`, 'resolve..') so in case a poweruser wants to build something custom it would be straight-forward already.
Thus maybe adding more arguments to to function is a bad idea as it will increase complexity for non power users

Reply:
"One of the crucial steps of the matching are matching the features from the results_df to the smbl_dfs species ids." Yeah, I struggled a bit with how to allow for different identifier matching approaches (like a table of {ontology, identifier}) tuples while still providing a high-level function. I'm planning to fold this bind_wide_results into an even higher level function which can also handle the process of passing attributes to the network (since this syntax is quite cryptic). I think the wide identifiers format is probably the one that we're most likely to see in the wild especially if metadata is coming from a table with one row per feature.